rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -163,6 +163,10 @@ BagMetadata MetadataIo::read_metadata(const std::string & uri)
     YAML::Node yaml_file = YAML::LoadFile(get_metadata_file_name(uri));
     auto metadata = yaml_file["rosbag2_bagfile_information"].as<rosbag2_storage::BagMetadata>();
     metadata.bag_size = FilesystemHelper::calculate_directory_size(uri);
+    // Preserve backward compatibility with older metadata files
+    if (metadata.version == 3) {
```

Comment:
As discussed let's check if the compression keys exist instead of versioning the file